pipeline/api/run/__init__.py

```diff
@@ -15,13 +10,13 @@ def run_pipeline(
     pipeline_id_or_schema: Union[str, PipelineGet], data_or_file_id: Union[Any, str]
 ):
     # TODO: Add support for generic object inference. Only strs at the moment.
-    file_id = None
-    if isinstance(data_or_file_id, str):
-        file_id = data_or_file_id
-    else:
-        temp_file = io.BytesIO(python_object_to_hex(data_or_file_id).encode())
-        uploaded_data = upload_file(temp_file, "/")
-        file_id = uploaded_data.id
+    # file_id = None
+    # if isinstance(data_or_file_id, str):
+    #     file_id = data_or_file_id
+    # else:
+    #     temp_file = io.BytesIO(python_object_to_hex(data_or_file_id).encode())
+    #     uploaded_data = upload_file(temp_file, "/")
+    #     file_id = uploaded_data.id
```

Comment:
Is this block just plain broken atm?

Reply:
unused

Reply:
I guess `file_id` is unused but I could imagine `upload_file` has side effects. Are you sure it's OK to remove it?

Reply:
it posts the file to api and returns the response

- but this relates to the sort of issue we should tackle, we can't be concerned with commenting a function in case its got side effects happening that we are not aware of!! in any case, I just uncommented and printed the var to skip the error quickly as having the code pristine is not the goal of this PR, but just adding the checks

Reply:
> we can't be concerned with commenting a function in case its got side effects happening that we are not aware of

In this case I think there was confusion with the fact that `file_id` was unused, which led to the whole block being commented without an understanding of the underlying intent. This is something we can improve on with clearer docs a broader test coverage, rather than an intrinsic problem with the structure of the code.

Agreed in general that there's a lot we can do to make the developer experience much better here.